src/snapred/backend/data/LocalDataService.py

```diff
@@ -540,17 +545,20 @@ def writeCalibrationWorkspaces(self, record: CalibrationRecord):
         -- assumes that `writeCalibrationRecord` has already been called, and that the version folder exists
         """
         calibrationDataPath = Path(self._constructCalibrationDataPath(record.runNumber, record.version))
-        calibrationTable = None
-        maskWorkspace = None
-        for workspace in record.workspaceNames:
-            if self.workspaceIsInstance(workspace, ITableWorkspace):
-                calibrationTable = workspace
-                continue
-            if self.workspaceIsInstance(workspace, MaskWorkspace):
-                maskWorkspace = workspace
-                continue
-            workspaceFilename = Path(workspace + ".nxs")
-            self.writeWorkspace(calibrationDataPath, workspaceFilename, workspace)
+        calibrationTable = ""
+        maskWorkspace = ""
+        # Assumes all workspaces are of WNG-type:
+        for wsKey, wsName in record.workspaces.items():
+            match wsKey:
+                case wngt.DIFFCAL_TABLE:
+                    calibrationTable = wsName
+                case wngt.DIFFCAL_MASK:
+                    maskWorkspace = wsName
```

Comment:
would 
```suggestion
        calibrationTable = record.workspaces.pop(wngt.DIFFCAL_TABLE, "")
        maskWorkspace = record.workspaces.pop(wngt.DIFFCAL_MASK, "")
        outputWorkspace = record.workspaces.pop(wngt.DIFFCAL_OUTPUT, "")
        
        if len(record.workspaces) > 0:
           ValueError... 
        
        if outputWorkspace:
            filename = ...
        ...
```
work better?

Reply:
Same problem as above -- there's now a possible list of workspace names at each key value.  However, the current solution also _assumes_ that there's only one 'table' + 'mask' combination -- definitely not an ideal approach.  **We really need a WNG name parser.**

Reply:
Right but the loop on a map and case statements is effectively the same as just accessing each case, defaulting to empty.
Its removes a layer of nesting.
If the argument is that is is purely temporary and we shouldnt dwell on it then I concede.

Reply:
**I actually ended up using your approach, although I made a copy, so the input record wouldn't be modified.**  :)  Doing stuff like assuming there was only _one_ mask + table combination really bugged me, so now I'm stepping through the lists (and there can now be multiple copies, although I expect that will never happen except for DIFFCAL_OUTPUT.